### PR TITLE
use jq for historical displayVersion

### PR DESCRIPTION
**Pull request by ryantm**

### CODE REVIEW

File: pkgs/historical-modules/default.nix [hunk 1]
```diff
@@ -113,16 +113,14 @@ let
...
-      moduleJSON = fromJSON (builtins.unsafeDiscardStringContext (readFile module.outPath)) // {
-        inherit displayVersion;
-      };
     in
-    pkgs.writeText "replit-module-${moduleId}" (toJSON moduleJSON);
+    pkgs.stdenvNoCC.mkDerivation {
+      name = "replit-module-${moduleId}";
+      buildCommand = ''
+        set -x
+        ${pkgs.jq}/bin/jq '.displayVersion = "${displayVersion}"' < ${module} > $out
```
Comment (blast-hardcheese) [inline pkgs/historical-modules/default.nix:121]:
```suggestion
        ${pkgs.jq}/bin/jq --arg displayVersion "${displayVersion}" '.displayVersion = ''$displayVersion' < ${module} > $out
```

if we're doing this anyway. iiuc `''$` should translate to a `$` in the script that gets run. If that's the case, then the suggestion here will work.

Comment (ryantm) [reply pkgs/historical-modules/default.nix]: I'm relying on interpolating in displayVersion from the Nix string there, not escaping it.  We could add displayVersion from Nix into the environment of the mkDerivation builder if you wanted to do it your way, but this seems simpler.

Comment (blast-hardcheese) [reply pkgs/historical-modules/default.nix]: Yeah, I'm just trying to handle the case where `${displayVersion}` contains something unexpected that results in not-json being emitted. Unlikely, but this is why `--arg` exists, so we can be certain. It's such a small number of entries though so it should be fine.

Comment (ryantm) [reply pkgs/historical-modules/default.nix]: Oh I see now cool!

File: pkgs/historical-modules/default.nix [hunk 2]
```diff
@@ -113,16 +113,14 @@ let
...
-      moduleJSON = fromJSON (builtins.unsafeDiscardStringContext (readFile module.outPath)) // {
-        inherit displayVersion;
-      };
     in
-    pkgs.writeText "replit-module-${moduleId}" (toJSON moduleJSON);
+    pkgs.stdenvNoCC.mkDerivation {
+      name = "replit-module-${moduleId}";
+      buildCommand = ''
+        set -x
+        ${pkgs.jq}/bin/jq '."displayVersion" = "${displayVersion}"' < ${module} > $out
```
Comment (blast-hardcheese) [inline pkgs/historical-modules/default.nix:121]:
```suggestion
        ${pkgs.jq}/bin/jq --arg displayVersion "${displayVersion}" '.displayVersion |= $displayVersion' < ${module} > $out
```
Tested locally:
```
$ nix build .#'"nodejs-16"' && jq < result | grep displayVersion
warning: Git tree '/app/nixmodules' is dirty
  "displayVersion": "16.18.1"
```

**Review by cdmistman — APPROVED** [review]
i also prefer using `jq --arg` but otherwise lgtm

don't forget pr description!

**Pull request merged**